### ISSUE TITLE
Can't import GitHub App service endpoint - resource requires one of auth_oauth, auth_personal

### ISSUE THREAD

**Issue by wunderhund** [microsoft/terraform-provider-azuredevops#326]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and Azure DevOps Provider) Version

```
$ terraform -v
Terraform v0.14.8
+ provider registry.terraform.io/hashicorp/azurerm v2.45.1
+ provider registry.terraform.io/microsoft/azuredevops v0.1.2
```

### Affected Resource(s)

* `azuredevops_serviceendpoint_github`

### Terraform Configuration Files

```
resource "azuredevops_serviceendpoint_github" "myendpoint" {
  project_id            = azuredevops_project.project.id
  service_endpoint_name = "myendpoint"
  # No auth_block should be required for an app service endpoint
  # Empty description = "" line no longer required thanks to #295 
}
```
As documented here: [azuredevops_serviceendpoint_github](https://registry.terraform.io/providers/microsoft/azuredevops/latest/docs/resources/serviceendpoint_github)

### Expected Behavior

Plan should succeed, or else another `auth_` setting should exist for GitHub Apps

### Actual Behavior

```
$ terraform plan
Acquiring state lock. This may take a few moments...
Releasing state lock. This may take a few moments...

Error: "auth_personal": one of `auth_oauth,auth_personal` must be specified

  on main.tf line 39, in resource "azuredevops_serviceendpoint_github" "myendpoint":
 39: resource "azuredevops_serviceendpoint_github" "myendpoint" {

Error: "auth_oauth": one of `auth_oauth,auth_personal` must be specified

  on main.tf line 39, in resource "azuredevops_serviceendpoint_github" "myendpoint":
 39: resource "azuredevops_serviceendpoint_github" "myendpoint" {
```

### Steps to Reproduce

1. Create a new Azure Service Connection of the `azure pipelines app` type in the Azure Devops GUI
     *  This can be done by creating a new pipeline, or by creating an OAuth service connection and attaching it to an existing pipeline in its `azuredevops_build_definition` block and/or its `azure-pipelines.yml` file. 
1. Get `id` of new service connection: `az devops service-endpoint list --org https://dev.azure.com/<myorg>/ -p <MYPROJECT>`
1. Create a new `azuredevops_serviceendpoint_github` resource with no `auth_` block
1. Import Service Connection to your resource: `$ terraform import azuredevops_serviceendpoint_github.myendpoint 00000000-0000-0000-0000-000000000000/00000000-0000-0000-0000-000000000000`
1. `terraform plan`

### Important Factoids

* Nothing unusual about account; running in `eastus`

* Seems likely these two Errors are triggered by these two `ExactlyOneOf:` checks:
  [resource_serviceendpoint_github.go#L40](https://github.com/microsoft/terraform-provider-azuredevops/blob/master/azuredevops/internal/service/serviceendpoint/resource_serviceendpoint_github.go#L40)
  [resource_serviceendpoint_github.go#L56](https://github.com/microsoft/terraform-provider-azuredevops/blob/master/azuredevops/internal/service/serviceendpoint/resource_serviceendpoint_github.go#L56)

* `terraform plan` will succeed if a dummy `auth_oauth` block is included, like so:
  ```
  auth_oauth {
    oauth_configuration_id = ""
  }
  ```
  However, if `terraform apply` is run with such a block in place, it will change the service endpoint type to OAuth and break it.

### References

* Similar but not duplicate issue: #295

**Comment by xuzhang3** [comment]
Hi @wunderhund GitHub service connection need at least one of `auth_oauth,auth_personal` specified.
 You can authorize with `oAuth` or `personal access token`, this is on your choice. If  the existed GitHub service connection  is authorized with `oAuth`, you need write your configuration like this:
```hcl
resource "azuredevops_serviceendpoint_github" "github_serviceendpoint" {
  project_id            = data.azuredevops_project.project.id
  service_endpoint_name = "GitHub Test"

  auth_oauth{
    # Also can be set with AZDO_GITHUB_SERVICE_CONNECTION_PAT environment variable
    oauth_configuration_id= "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
  }
}
```

otherwise:
```hcl
resource "azuredevops_serviceendpoint_github" "github_serviceendpoint" {
  project_id            = data.azuredevops_project.project.id
  service_endpoint_name = "GitHub Test"

 auth_personal {
    # Also can be set with AZDO_GITHUB_SERVICE_CONNECTION_PAT environment variable
    personal_access_token = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
  }
}
```

**Comment by wunderhund** [comment]
@xuzhang3 yes, GitHub service connection need at least one of `auth_oauth, auth_personal` specified. **That is the problem.** The Azure API contains a third authorization scheme for Service Connections configured through Github Apps. In the Azure API it looks like this:

```
az devops service-endpoint show --org https://dev.azure.com/myorg/ -p MYPROJECT --id 00000000-0000-0000-0000-00000000000
<...>
    "authorization": {
      "parameters": null,
      "scheme": "InstallationToken"
    },
<...>
```
This is what it looks like in the Azure portal:
<img width="236" alt="Using Azure Pipelines App" src="https://user-images.githubusercontent.com/7594832/111310062-4041db80-8622-11eb-8258-0b525513d971.png">

If you have a PAT or OAuth service connection, Azure portal will prompt you to convert it to a Github Apps service connection type:
<img width="605" alt="Switch to Azure Pipelines Github App" src="https://user-images.githubusercontent.com/7594832/111310055-3e781800-8622-11eb-8916-bd021357f860.png">

In order to be able to properly import these service connection types, we need to be able to make `azuredevops_serviceendpoint_github` resource blocks _without_ the `auth_oauth, auth_personal` restriction. _Or_, we need a new authorization block type to account for this.

**Comment by xuzhang3** [comment]
Hi @wunderhund  I submit a PR #328 to fix this issue, thanks for your feedback.

**Comment by xuzhang3** [comment]
Fixed in v0.1.3, closing this issue. Feel free to open another issue if you have questions.
